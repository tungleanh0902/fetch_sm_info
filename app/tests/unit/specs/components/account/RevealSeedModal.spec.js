import { shallowMount, createLocalVue } from "@vue/test-utils"
import Vuelidate from "vuelidate"
import VueClipboard from "vue-clipboard2"
import RevealSeedModal from "src/components/account/RevealSeedModal"

const localVue = createLocalVue()
localVue.use(Vuelidate)
localVue.directive(`clipboard`, VueClipboard)

jest.mock(`src/../config.js`, () => ({
  isExtension: false,
}))

const $route = {
  params: {
    address: `cosmos1234`,
  },
}

describe(`RevealSeedModal`, () => {
  let wrapper, $store

  beforeEach(() => {
    $store = {
      dispatch: async () => {
        return { seedPhrase: `seed1 seed2 seed3` }
      },
    }
    wrapper = shallowMount(RevealSeedModal, {
      localVue,
      mocks: {
        $route,
        $store,
      },
    })
  })

  it(`should show the RevealSeedModal page`, async () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`should return the wallet secret, whether a seed or a private key`, async () => {
    let self = {
      $route,
      wallet: {
        seedPhrase: `abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon art`,
      },
    }
    let secret = RevealSeedModal.computed.seedOrPrivateKey.call(self)
    expect(secret).toBe(
      `abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon art`
    )
    self = {
      $route,
      wallet: {
        privateKey: `0123456789`,
      },
    }
    secret = RevealSeedModal.computed.seedOrPrivateKey.call(self)
    expect(secret).toBe(`0123456789`)
    self = {
      $route,
      wallet: {},
    }
    secret = RevealSeedModal.computed.seedOrPrivateKey.call(self)
    expect(secret).toBe(``)
  })

  it(`should set recoveryError to true when failing to fetch wallet from stored wallets`, async () => {
    $store = { dispatch: jest.fn() }
    wrapper = shallowMount(RevealSeedModal, {
      localVue,
      mocks: {
        $route,
        $store,
      },
    })
    wrapper.setData({
      password: `password`,
    })
    await wrapper.vm.revealSeedPhrase()
    expect(wrapper.vm.recoveryError).toBe(true)
  })

  it(`should get the wallet and set it in data if it contains the secret (seed or private key)`, async () => {
    wrapper.setData({
      password: `password`,
    })
    await wrapper.vm.revealSeedPhrase()
    expect(wrapper.vm.wallet).toEqual({ seedPhrase: `seed1 seed2 seed3` })
  })

  it(`should set input type for password input as "text" or as "password" if it is already "text".
        It should also increase by one this component key`, () => {
    wrapper.setData({
      passwordInputType: `text`,
      passwordInputKey: 0,
    })
    wrapper.vm.showPassword()
    expect(wrapper.vm.passwordInputType).toBe(`password`)
    expect(wrapper.vm.passwordInputKey).toBe(1)

    wrapper.setData({
      passwordInputType: `hey`,
    })
    wrapper.vm.showPassword()
    expect(wrapper.vm.passwordInputType).toBe(`text`)
    expect(wrapper.vm.passwordInputKey).toBe(2)
  })
})
