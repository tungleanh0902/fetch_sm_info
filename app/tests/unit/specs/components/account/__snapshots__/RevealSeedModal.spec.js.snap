// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RevealSeedModal should show the RevealSeedModal page 1`] = `
<sessionframe-stub
  icon=""
>
  <div
    class="session-container"
  >
    <h2
      class="title reveal-title"
    >
      
      You are about to reveal
      <br />
       
      <span
        class="pill"
      >
        your seed phrase.
      </span>
    </h2>
     
    <tmformgroup-stub
      class="reveal-seed-form-group"
      fieldid="password"
    >
      <tmfield-stub
        class="passwordInput"
        id="password"
        placeholder="Password"
        type="password"
        value=""
      />
       
      <div
        class="reveal-seed-show-password"
      >
        <i
          class="material-icons notranslate"
        >
          visibility
        </i>
      </div>
       
      <!---->
       
      <!---->
       
      <!---->
       
      <div
        class="reveal-seed-buttons"
      >
        <tmbtn-stub
          class="reveal-seed-button"
          customclass=""
          type="secondary"
          value="Dismiss"
        />
         
        <tmbtn-stub
          class="reveal-seed-button"
          customclass=""
          type="primary"
          value="Reveal"
        />
      </div>
    </tmformgroup-stub>
  </div>
</sessionframe-stub>
`;
